synthesizer/program/src/resources/credits.aleo

```diff
@@ -12,6 +12,38 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/********************************************** INVARIANTS ************************************************************/
+
+// 1. contains committee[r0] =>
+//   a. contains delegated[r0]
+//      - The set of delegated addresses is superset of active validators.
+// 2. delegated[*] >= 10,000
+//   a. contain bonded[*].microcredits >= 10,000
+//      - Delegators can bond to any address with the 10,000 credit minimum
+// 3. delegated[r0] < 10,000,000 =>
+//   a. !contain committee[r0]
+//      - If the delegated total (which includes the self bond) is less than 10,000,000 credits, it won't be in the committee
+// 4. bond[r0].validator == r0 =>
+//   a. contains committee[r0]
+//     - Self bonded addresses are always active validators. Delegators **can** force validators to unbond by removing sufficient delegation
+// 5. committee[r0] =>
+//   a. bonded[r0].microcredits >= 100 credits
+//      - Self bonded addresses are always active validators with at least 100 credits self bonded
+//   b. delegated[r0] >= 10,000,000 credits
+//      - The total delegated amount for an active validator is at least 10,000,000 credits
+// 6. delegated[r0] == sum bond[*].microcredits by bond[].validator == r0
+// 7. bond[r0].microcredits < 10,000 =>
+//   a. bond[r0].validator == r0
+//   b. contains committee[r0]
+//   - A bond value < 10,000 implies self-bonded active validator.
+// 8. bond[r0].validator != r0 =>
+//   a. bond[r0].microcredits >= 10,000
+// 9. count bonded[] - count committee[] == metadata[aleo1qgqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqanmpl0]
+// 10. count committee[] == metadata[aleo1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq3ljyzc]
+// 11. 100,000 >= count bonded[] >= count delegated[] >= count committee[]
```

Comment:
Is this correct?

The maximum number of delegators is 100,000. However, the `count bonded` is the sum of `count delegated` and `count committee`.

So `count bonded` could be `100,000 + number of validators` which is greater than `100,000`.

Reply:
You're right, I'll update it.